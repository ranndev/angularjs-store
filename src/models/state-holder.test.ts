import StateHolder, { create } from './state-holder';

const state = { foo: '', bar: 1, baz: false };
let stateHolder: StateHolder<typeof state>;

beforeEach(() => {
  stateHolder = create(state);
});

describe('StateHolder', () => {
  it('should match the initial state to snapshot', () => {
    expect(stateHolder).toMatchSnapshot();
  });

  describe('get', () => {
    it('should always return a new copy of state', () => {
      const copies: Array<typeof state> = [];
      for (let i = 0; i < 9; i++) {
        const copy = stateHolder.get() as typeof state;
        expect(copies).not.toContain(copy);
        copies.push(copy);
      }
    });

    it('should return a specific property state', () => {
      Object.entries(state).forEach(([key, value]) => {
        expect(value).toBe(stateHolder.get(key as keyof typeof state));
      });
    });
  });

  describe('set', () => {
    it.each<[Partial<typeof state>]>([
      [{ foo: 'bar' }],
      [{ foo: 'baz', bar: 100 }],
      [{ foo: 'fuz', bar: 200, baz: true }],
    ])('should support updating state partially', (partialState) => {
      stateHolder.set(partialState);
      expect(stateHolder.get()).toEqual({ ...state, ...partialState });
    });
  });
});
