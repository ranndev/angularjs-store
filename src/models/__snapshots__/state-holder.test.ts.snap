// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateHolder should match the initial state to snapshot 1`] = `
Object {
  "get": [Function],
  "set": [Function],
}
`;
